docs: add Control keys and control characters guide (#2249)
keymanapp/help.keyman.com: developer/language/guide/control-keys.md

```
+ 's' > $medialSigma

c Transform medial sigma at end of word to final sigma
$medialSigma + [K_SPACE] > $finalSigma ' '

c If Enter key is pressed after a medial sigma, transform that to
c a final sigma, then pass the Enter key on to the application
$medialSigma + [K_ENTER] > $finalSigma use(emit)

c ...

group(emit) using keys
c this empty group will emit the original keystroke
```

## Backspace

It can also be helpful in Keyman to handle the <kbd>Backspace</kbd> key in a
number of situations: decomposing characters, reordering, or deleting multiple
components together. Keyman allows this with rules matching `[K_BKSP]`.

For example, in Khmer, subscript consonants are encoded with a subscript
("coeng") marker `U+17D2`, followed by the consonant character. The [Khmer
Angkor keyboard](https://github.com/keymanapp/keyboards/blob/master/release/k/khmer_angkor/source/khmer_angkor.kmn)
treats these two characters as a single element in its touch keyboard, so the
user will never even see the coeng marker character. When <kbd>Backspace</kbd>
is pressed, the keyboard deletes both characters together:

```
platform('touch') U+17D2 any(consonant) + [K_BKSP] > nul
```

> The <kbd>Backspace</kbd> key at the start of a text block in a complex document

Reviewer: I don't understand this first sentence. Is there an `or` or something missing? Or what does deleting the text block have to do with deleting a bullet from a bullet list?

Author: ```suggestion
Pressing the <kbd>Backspace</kbd> key at the start of a text block in a complex document
```

> may delete the text block, for example, it may delete a bullet from in a bullet

Reviewer: ```suggestion
may delete the text block, for example, it may delete a bullet from a bullet
```

Author: ```suggestion
may delete the break or formatting before the text block; 
for example, it may delete a bullet from a bullet
```

list. You should avoid overriding this behavior, which is application-specific.

```
nul + [K_BKSP] > nul     c DO NOT DO THIS
+ [K_BKSP] > nul         c DO NOT DO THIS EITHER
```

# Control characters in output

> It is best to never to emit control characters from a Keyman rule. Applications

Reviewer: ```suggestion
It is best to never emit control characters from a Keyman rule. Applications
```

In a Polytonic Greek keyboard, you may wish to automatically insert breathing
marks at the start of words. In a touch keyboard, you may want to [recognize the
start of a sentence](casing-support) and switch to a Shift layer.

However, you do need to be careful here. When recognizing the control character
in the context, it is easy to accidentally emit the control character in the
output. For example, the following rule in a Hindi keyboard may appear harmless,
but actually deletes and re-inserts the LF character, which can be problematic
in some applications:

```
  c linebreak, add full svara.
  U+000A U+094D > U+000A U+0905 c a    c DO NOT DO THIS
```

Instead, we can make use of how Keyman applies the [`context`](../reference/context)
statement. When Keyman encounters the `context` statement at the start of the
output, it does not delete and re-insert the characters found there. So we can
process the required character change in a separate group, without accidentally
touching the control character:

```
  U+000A U+094D > context use(full-svara-after-lf)

  group(full-svara-after-lf)

  c here we know we that we have an LF immediately prior
  U+094D > U+0905 c a
```

> This behavior is present from v6.0 through v18.0 of Keyman, but may be improved

Reviewer: I see that you marked #14718 for v20, so I'd omit mentioning v18.0 here:

```suggestion
This behavior is present from v6.0 of Keyman, but may be improved
```
